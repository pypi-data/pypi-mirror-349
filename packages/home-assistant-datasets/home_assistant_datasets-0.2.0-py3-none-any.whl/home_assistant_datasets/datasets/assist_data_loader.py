"""Module for reading 'assist' dataset file records.

The Assist dataset format was originally based on the home assistant intents
test format, but modified to perform more fine grained setup and testing of
entity states. This is better than just expecting a tool call since it can
deal with fuzzy actions that result in the same outcome.

An individual record contains sentences to scrape the model with. Each sentence
is expanded into it's own `EvalTask`.
"""

from collections.abc import Generator
import datetime
from dataclasses import dataclass, field
import logging
import pathlib
from slugify import slugify
from typing import Any

from mashumaro.mixins.yaml import DataClassYAMLMixin
from mashumaro.config import BaseConfig
from mashumaro.exceptions import MissingField

from home_assistant_datasets.entity_state import EntityState
from home_assistant_datasets.yaml_loaders import yaml_decode

from .dataset_card import DatasetCard

__all__ = [
    "Record",
    "Action",
    "read_record",
]

_LOGGER = logging.getLogger(__name__)


@dataclass(kw_only=True, frozen=True)
class ToolCall(DataClassYAMLMixin):
    """Represents an expected tool call."""

    tool_name: str
    """The name of the tool to call."""

    tool_args: dict[str, Any] | None = None
    """Arguments to the tool call."""

    class Config(BaseConfig):
        code_generation_options = ["TO_DICT_ADD_OMIT_NONE_FLAG"]
        forbid_extra_keys = True
        sort_keys = False
        omit_none = True


@dataclass(kw_only=True, frozen=True)
class Action(DataClassYAMLMixin):
    """An individual data item action."""

    sentences: list[str]
    """Sentences spoken."""

    setup: dict[str, EntityState] = field(default_factory=dict)
    """Initial entity states to override keyed by entity_id."""

    expect_changes: dict[str, EntityState] | None = None
    """The device states to assert on, keyed by entity_id."""

    ignore_changes: dict[str, list[str]] | None = None
    """The device state or attribute changes to ignored, keyed by entity_id."""

    expect_response: str | list[str] | None = None
    """Expect the agent to respond with this substring.

    When specified as a list, the response may match any valid substring in the last.
    """

    expect_tool_call: ToolCall | None = None
    """Expect the specified tool call to occur, independent of the outcome."""

    context_device: str | None = None
    """Synthetic home device id for the current context of the request."""

    context_now: datetime.datetime | None = None
    """The current time to use during tests."""

    class Config(BaseConfig):
        code_generation_options = ["TO_DICT_ADD_OMIT_NONE_FLAG"]
        forbid_extra_keys = False
        sort_keys = False
        omit_none = True
        omit_default = True


@dataclass(kw_only=True, frozen=True)
class RecordSource(DataClassYAMLMixin):
    """Information about the Record file source."""

    record_id: str
    """A unique identifier of this record within the dataset."""

    record_path: pathlib.Path
    """File path for this dataset record."""


@dataclass(kw_only=True)
class Record(DataClassYAMLMixin):
    """Represents an item in the dataset used to configure evaluation."""

    category: str | list[str]
    """Category used to describe the evaluation task when reporting"""

    tests: list[Action] | None = field(default_factory=list)
    """The set of tasks to evaluate."""

    record_source: RecordSource | None = None
    """Information about the filesystem source for this record.

    Attribute is set at runtime after being loaded.
    """

    @property
    def categories(self) -> list[str]:
        """Labels used for slicing model predictions."""
        if isinstance(self.category, list):
            return self.category
        return [self.category]

    class Config(BaseConfig):
        code_generation_options = ["TO_DICT_ADD_OMIT_NONE_FLAG"]
        sort_keys = True
        omit_none = True


def _make_slug(text: str) -> str:
    """Shorthand slugify command"""
    return slugify(text, separator="_")


def read_record_sources(dataset_card: DatasetCard) -> Generator[RecordSource]:
    """Generate the list of dataset files for the given path."""

    def unique_record_id(filename: pathlib.Path) -> str:
        """Convert the filename to a unique id withint he dataset."""
        assert dataset_card.path is not None
        relative_path = filename.relative_to(dataset_card.dataset_path)
        stem = str(relative_path)[:-5]  # No .yaml suffix
        return _make_slug(stem)

    for filename in dataset_card.dataset_files:
        yield RecordSource(record_id=unique_record_id(filename), record_path=filename)


def read_record(filename: pathlib.Path) -> Record:
    """Read the dataset record"""
    try:
        return yaml_decode(filename.open(), Record)
    except MissingField as err:
        raise ValueError(f"Dataset record invalid {filename}: {err}") from err


def read_dataset_records(
    dataset_card: DatasetCard, categories: set[str] | None = None
) -> Generator[Record]:
    """Generate the list of dataset records for the given path."""

    for record_source in read_record_sources(dataset_card):
        assert record_source.record_path is not None
        record = read_record(record_source.record_path)
        if categories and not any(
            category in categories for category in record.categories
        ):
            _LOGGER.debug(
                "Skipping record with category %s (not in %s)",
                record.category,
                categories,
            )
            continue
        record.record_source = record_source
        yield record
