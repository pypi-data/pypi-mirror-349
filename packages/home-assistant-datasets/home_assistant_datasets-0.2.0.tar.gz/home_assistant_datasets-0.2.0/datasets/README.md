# Datasets

This directory contains synthetic data used for generating home information.
The datasets in this repository are generated using the notebooks and seeds in
the `../generation` directory.

## Homes

Home data is generated using the `synthetic-home` notebook. These homes are
used as a basis to generate other downstream data such as areas, devices,
entities, and values.

Example data:

```
homes:
  - name: "Cozy Cottage"
    country_code: "US"
    location: "Rural area in Maine"
    type: "Single-family house"
    amenities:
      - 2 bedrooms
      - 1 bathroom
      - Open concept living room and kitchen
      - Front porch with swing
      - Fire pit in the backyard
      - Detached garage
```

## Area devices

This is created by reading the homes data and creating areas and devices for the
home. See `synthetic-devices` notebook.

The data is a single yaml file with many documents in it (since that was easier)
to create from the data generation step. The dataset contains the chain of
thought used to create the areas and devices.
```
---
...
areas:
  thoughts:
  - The house is in a suburban neighborhood in Munich, so the climate may vary throughout
    the year.
  - The front yard with a vegetable garden suggests possible smart irrigation for
    plants.
  - With 4 bedrooms, there may be multiple family members living in the house.
  - The detached garage may have a smart garage door opener.
  - The kitchen with a breakfast nook could benefit from smart lighting for various
    uses.
  area_devices:
    Living Room:
    - light
    - thermostat
    - smart_tv
    Kitchen:
    - light
    - smart_light
    - smart_speaker
    Dining Area:
    - light
    - smart_light
    - smart_speaker
    Master Bedroom:
    - light
    - smart_tv
    Bedroom 2:
    - light
    - smart_speaker
    Front Yard:
    - light
    - sprinkler
    Garage:
    - cover
    - light
  other_devices:
  - smartphone
  - tablet
  - laptop
```

## Entities

The entities are created by `synthetic-device-entities` based on the devices,
with an additional dictonary for storing the entity information for each device.
The area information is repeated to make it easier to access.
```
device_entities:
- Living Room:
  - name: light
    entities:
    - light.living_room
  - name: thermostat
    entities:
    - climate.living_room
    - sensor.living_room_temperature
    - sensor.living_room_humidity
  - name: smart_tv
    entities:
    - media_player.living_room
- Kitchen:
  - name: light
    entities:
    - light.kitchen
  - name: smart_light
    entities:
    - light.kitchen_counter
    - light.kitchen_island
  - name: smart_speaker
    entities:
    - media_player.kitchen
- Dining Area:
  - name: light
    entities:
    - light.dining_area
  - name: smart_light
    entities:
    - light.dining_area_chandelier
  - name: smart_speaker
    entities:
    - media_player.dining_area
- Master Bedroom:
  - name: light
    entities:
    - light.master_bedroom
  - name: smart_tv
    entities:
    - media_player.master_bedroom_tv
    - remote.master_bedroom_tv
    - binary_sensor.master_bedroom_tv_headphones_connected
- Bedroom 2:
  - name: light
    entities:
    - light.bedroom_2
  - name: smart_speaker
    entities:
    - media_player.bedroom_2
- Front Yard:
  - name: light
    entities:
    - light.front_yard
  - name: sprinkler
    entities:
    - switch.front_yard_sprinkler
- Garage:
  - name: cover
    entities:
    - cover.garage_door
  - name: light
    entities:
    - light.garage
- Other:
  - name: smartphone
    entities:
    - device_tracker.iphone
  - name: tablet
    entities:
    - device_tracker.tablet
  - name: laptop
    entities:
    - device_tracker.laptop
```

## Actions

Synthetic actions for a home are created with `synthetic-actions` and appended
to the home record. The instructions are to generate actions that may or
may not be valid for the particular home.

```
actions:
- Summarize the current temperature and humidity levels in the living room
- Evaluate the energy usage in the home over the past week
- Turn on the air purifier in the bedroom if the air quality is poor
- Notify the user if the front door has been left open for more than 10 minutes
- Determine if any lights have been left on when the last person left the home
- Set the thermostat to an energy-saving mode when the home is unoccupied
- Notify the user if the refrigerator door has been open for an extended period
- Evaluate the water usage in the home during the last month
- Turn off all electronic devices in the home before bedtime
- Notify the user if the smart oven has finished preheating
- Determine if the garage door is closed when the last person leaves the home
- Evaluate the performance of the smart sprinkler system based on recent watering
  schedules
- Notify the user if the smart smoke detector detects smoke or fire
- Set the smart blinds to open at sunrise and close at sunset
- Determine if the smart security camera has detected any motion in the backyard
- Evaluate the performance of the smart thermostat in maintaining a comfortable temperature
- Notify the user if the smart speaker detects unusual sounds in the home
- Turn off the bathroom lights automatically if they have been left on for more than
  30 minutes
- Determine if the smart TV has been left on while nobody is in the living room
- Evaluate the energy efficiency of the smart dishwasher based on usage patterns
```
