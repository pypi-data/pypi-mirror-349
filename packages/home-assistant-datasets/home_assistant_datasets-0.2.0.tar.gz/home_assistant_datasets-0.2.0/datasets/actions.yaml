---
name: Family Home
country_code: DE
location: Suburban neighborhood in Munich
type: Single-family house
amenities:
- 4 bedrooms
- 2.5 bathrooms
- Spacious living room and dining area
- Kitchen with breakfast nook
- Front yard with vegetable garden
- Detached garage
areas:
  thoughts:
  - The house is in a suburban neighborhood in Munich, so the climate may vary throughout
    the year.
  - The front yard with a vegetable garden suggests possible smart irrigation for
    plants.
  - With 4 bedrooms, there may be multiple family members living in the house.
  - The detached garage may have a smart garage door opener.
  - The kitchen with a breakfast nook could benefit from smart lighting for various
    uses.
  area_devices:
    Living Room:
    - light
    - thermostat
    - smart_tv
    Kitchen:
    - light
    - smart_light
    - smart_speaker
    Dining Area:
    - light
    - smart_light
    - smart_speaker
    Master Bedroom:
    - light
    - smart_tv
    Bedroom 2:
    - light
    - smart_speaker
    Front Yard:
    - light
    - sprinkler
    Garage:
    - cover
    - light
  other_devices:
  - smartphone
  - tablet
  - laptop
device_entities:
- Living Room:
  - name: light
    entities:
    - light.living_room
  - name: thermostat
    entities:
    - climate.living_room
    - sensor.living_room_temperature
    - sensor.living_room_humidity
  - name: smart_tv
    entities:
    - media_player.living_room
- Kitchen:
  - name: light
    entities:
    - light.kitchen
  - name: smart_light
    entities:
    - light.kitchen_counter
    - light.kitchen_island
  - name: smart_speaker
    entities:
    - media_player.kitchen
- Dining Area:
  - name: light
    entities:
    - light.dining_area
  - name: smart_light
    entities:
    - light.dining_area_chandelier
  - name: smart_speaker
    entities:
    - media_player.dining_area
- Master Bedroom:
  - name: light
    entities:
    - light.master_bedroom
  - name: smart_tv
    entities:
    - media_player.master_bedroom_tv
    - remote.master_bedroom_tv
    - binary_sensor.master_bedroom_tv_headphones_connected
- Bedroom 2:
  - name: light
    entities:
    - light.bedroom_2
  - name: smart_speaker
    entities:
    - media_player.bedroom_2
- Front Yard:
  - name: light
    entities:
    - light.front_yard
  - name: sprinkler
    entities:
    - switch.front_yard_sprinkler
- Garage:
  - name: cover
    entities:
    - cover.garage_door
  - name: light
    entities:
    - light.garage
- Other:
  - name: smartphone
    entities:
    - device_tracker.iphone
  - name: tablet
    entities:
    - device_tracker.tablet
  - name: laptop
    entities:
    - device_tracker.laptop
actions:
- Summarize the current temperature and humidity levels in the living room
- Evaluate the energy usage in the home over the past week
- Turn on the air purifier in the bedroom if the air quality is poor
- Notify the user if the front door has been left open for more than 10 minutes
- Determine if any lights have been left on when the last person left the home
- Set the thermostat to an energy-saving mode when the home is unoccupied
- Notify the user if the refrigerator door has been open for an extended period
- Evaluate the water usage in the home during the last month
- Turn off all electronic devices in the home before bedtime
- Notify the user if the smart oven has finished preheating
- Determine if the garage door is closed when the last person leaves the home
- Evaluate the performance of the smart sprinkler system based on recent watering
  schedules
- Notify the user if the smart smoke detector detects smoke or fire
- Set the smart blinds to open at sunrise and close at sunset
- Determine if the smart security camera has detected any motion in the backyard
- Evaluate the performance of the smart thermostat in maintaining a comfortable temperature
- Notify the user if the smart speaker detects unusual sounds in the home
- Turn off the bathroom lights automatically if they have been left on for more than
  30 minutes
- Determine if the smart TV has been left on while nobody is in the living room
- Evaluate the energy efficiency of the smart dishwasher based on usage patterns
---
name: City Loft
country_code: US
location: Urban area in New York City
type: Apartment
amenities:
- Studio layout
- 1 bathroom
- Floor-to-ceiling windows with city views
- Shared rooftop garden
- Fitness center in building
- Laundry in unit
areas:
  thoughts:
  - The studio layout suggests a single occupant.
  - The floor-to-ceiling windows with city views may benefit from a smart light for
    ambiance.
  - The shared rooftop garden may not allow for personal smart devices.
  - Having laundry in-unit may not require a smart washer/dryer setup.
  - The fitness center in the building may not necessitate personal home gym devices.
  area_devices:
    Living Area:
    - light
    - smart_tv
    Bathroom:
    - light
    - speaker
    Laundry Area:
    - light
  other_devices:
  - smartphone
device_entities:
- Living Area:
  - name: light
    entities:
    - light.living_area
  - name: smart_tv
    entities:
    - media_player.living_area_tv
    - remote.living_area_tv
    - binary-sensor.living_area_tv_headphones_connected
- Bathroom:
  - name: light
    entities:
    - light.bathroom
  - name: speaker
    entities:
    - media_player.bathroom_speaker
- Laundry Area:
  - name: light
    entities:
    - light.laundry_area
- Other:
  - name: smartphone
    entities:
    - device_tracker.iphone
actions:
- Summarize the current status of the kitchen devices
- Evaluate the energy consumption in the bedroom over the past week
- Turn off the lights in the living room if no motion is detected for 30 minutes
- Notify the user if the front door has been left unlocked for more than 1 hour
- Determine if the temperature in the nursery is outside the desired range
- Check if the humidity level in the bathroom is too high
- Analyze the water usage pattern in the laundry room
- Notify the user if the garage door has been open for longer than usual
- Determine if the oven has been left on for an extended period
- Evaluate the air quality in the dining room
- Check the status of the smoke detectors in the hallway
- Determine if the blinds in the study are closed during the day
- Notify the user if any windows are left open when the heating is on
- Analyze the sound levels in the entertainment room during movie nights
- Determine if the smart thermostat is optimizing energy usage efficiently
- Check the battery levels of all the smart devices in the home
- Notify the user if the smart security cameras detect unusual activity
- Evaluate the performance of the smart sprinkler system in the garden
- Determine if the smart TV in the living room is being used excessively
- Analyze the trends in temperature fluctuations across different rooms
---
name: Home5
country_code: RU
location: Industrial district in Novosibirsk, Russia
type: Loft
amenities:
- Studio layout
- 1 bathroom
- High ceilings with exposed beams
- Modern kitchen
- Large windows with city views
- Secured parking lot
areas:
  thoughts:
  - The loft in an industrial district may have a modern and industrial aesthetic.
  - The studio layout suggests a single occupant or a couple living in the loft.
  - The large windows with city views may benefit from smart blinds or curtains.
  - The secured parking lot may not require a smart garage door opener if access is
    already restricted.
  area_devices:
    Living Area:
    - light
    - smart_tv
    Kitchen:
    - light
    - smart_appliances
    Bathroom:
    - light
    - smart_mirror
    Windows:
    - smart_blinds
  other_devices:
  - smartphone
  - tablet
  - laptop
  - security_camera
device_entities:
- Living Area:
  - name: light
    entities:
    - light.living_area
  - name: smart_tv
    entities:
    - media_player.living_area_tv
    - remote.living_area_tv
    - binary_sensor.living_area_tv_headphones_connected
- Kitchen:
  - name: light
    entities:
    - light.kitchen
  - name: smart_appliances
    entities:
    - switch.kitchen_dishwasher
    - switch.kitchen_oven
    - sensor.kitchen_fridge_temperature
    - sensor.kitchen_freezer_temperature
- Bathroom:
  - name: light
    entities:
    - light.bathroom
  - name: smart_mirror
    entities:
    - switch.bathroom_mirror_lights
    - sensor.bathroom_mirror_temperature
    - sensor.bathroom_mirror_humidity
- Windows:
  - name: smart_blinds
    entities:
    - cover.windows_blinds
- Other:
  - name: smartphone
    entities:
    - device_tracker.iphone
  - name: tablet
    entities:
    - device_tracker.tablet
  - name: laptop
    entities:
    - device_tracker.laptop
  - name: security_camera
    entities:
    - camera.outdoor_camera
actions:
- Summarize the current activities in the kitchen area
- Evaluate the air quality in the bedroom
- Turn off the smart light bulbs in the living room
- Notify the user if the front door has been left open for more than 10 minutes
- Determine if the temperature in the bathroom is comfortable
- Alert the user if the garage door has been open for more than an hour
- Check if the smart thermostat is set to an energy-saving mode
- Notify the user if the humidity levels in the nursery are too high
- Determine if the window blinds in the study room are open or closed
- Evaluate the energy consumption in the home office
- Notify the user if the laundry cycle has been completed
- Check if the smart TV in the entertainment room is turned on
- Determine if the motion sensors have detected any movement in the hallway
- Notify the user if the water heater has been running for an extended period
- Evaluate the noise levels in the home during the night
- Check if the smart locks have been tampered with
- Determine if the smart sprinkler system needs to be activated
- Notify the user if the refrigerator door has been left open
- Evaluate the lighting conditions in the dining area
- Check if the security cameras have detected any unusual activity
---
name: Family Farmhouse
country_code: US
location: Rural area in Iowa
type: Farmhouse
amenities:
- 4 bedrooms
- 3 bathrooms
- Large family room with fireplace
- Farmhouse kitchen with pantry
- Wrap-around porch
- Barn and chicken coop in the backyard
areas:
  thoughts:
  - The farmhouse is located in a rural area, so there may be a need for security
    cameras around the property.
  - The presence of a barn and chicken coop suggests the possibility of a smart feeding
    system for the animals.
  - The large family room with a fireplace may benefit from a smart thermostat for
    temperature control.
  - The wrap-around porch offers an opportunity for smart outdoor lighting.
  area_devices:
    Family Room:
    - light
    - smart_tv
    - thermostat
    Kitchen:
    - light
    - smart_oven
    Pantry:
    - light
    Backyard:
    - camera
    Barn:
    - smart_feeder
    Chicken Coop:
    - smart_feeder
    Master Bedroom:
    - light
    - smart_blinds
  other_devices:
  - smartphone 1
  - smartphone 2
  - tablet
  - laptop
device_entities:
- Family Room:
  - name: light
    entities:
    - light.family_room
  - name: smart_tv
    entities:
    - media_player.family_room_tv
    - remote.family_room_tv
    - binary-sensor.family_room_tv_headphones_connected
  - name: thermostat
    entities:
    - climate.family_room
    - sensor.family_room_temperature
    - sensor.family_room_humidity
- Kitchen:
  - name: light
    entities:
    - light.kitchen
  - name: smart_oven
    entities:
    - switch.kitchen_oven
    - sensor.kitchen_oven_temperature
- Pantry:
  - name: light
    entities:
    - light.pantry
- Backyard:
  - name: camera
    entities:
    - camera.backyard
- Barn:
  - name: smart_feeder
    entities:
    - switch.barn_feeder
- Chicken Coop:
  - name: smart_feeder
    entities:
    - switch.chicken_coop_feeder
- Master Bedroom:
  - name: light
    entities:
    - light.master_bedroom
  - name: smart_blinds
    entities:
    - switch.master_bedroom_blinds
- Other:
  - name: smartphone 1
    entities:
    - device_tracker.smartphone_1
  - name: smartphone 2
    entities:
    - device_tracker.smartphone_2
  - name: tablet
    entities:
    - device_tracker.tablet
  - name: laptop
    entities:
    - device_tracker.laptop
actions:
- Summarize the current temperature and humidity levels in the bedroom
- Evaluate the energy usage in the kitchen over the past week
- Turn off all smart plugs in the home
- Notify the user if the front door has been left open for more than 10 minutes
- Determine if the kitchen lights have been on for more than 3 hours
- Evaluate the air quality in the living room and suggest actions if necessary
- Notify the user if the garage door has been open for longer than usual
- Determine if any windows are left open and suggest closing them
- Summarize the weather forecast for the next 3 days
- Notify the user if the thermostat is set too high for energy efficiency
- Evaluate the water usage in the bathroom over the past month
- Turn on the porch lights when the sun sets
- Notify the user if the smart water heater is malfunctioning
- Determine if the home security system is armed before bedtime
- Evaluate the noise levels in the home office during working hours
- Notify the user if the refrigerator door has been left ajar
- Determine if the smart blinds need maintenance or cleaning
- Evaluate the motion sensor activity in the hallway over the past week
- Notify the user if the smart smoke detector battery is low
- Determine if the air purifier needs filter replacement
---
name: "Mas Proven\xE7al"
country_code: FR
location: Vineyard in Provence, France
type: Traditional Provencal farmhouse
amenities:
- "6 chambres d\xE9cor\xE9es dans le style proven\xE7al"
- 4 salles de bains avec baignoire en fonte
- "Salle \xE0 manger spacieuse avec chemin\xE9e en pierre"
- "Cuisine campagnarde avec four \xE0 pain"
- "Cour int\xE9rieure avec piscine et pool house"
- "Terrain arbor\xE9 d'oliviers et de lavande"
- "Caves \xE0 vin et d\xE9pendances"
areas:
  thoughts:
  - The traditional farmhouse in Provence may have a classic and rustic interior design.
  - The spacious dining room with a stone fireplace suggests a cozy and inviting atmosphere.
  - The farmhouse's courtyard with a pool and pool house is a luxurious feature, perfect
    for relaxation.
  - The olives and lavender trees in the garden indicate a focus on outdoor living
    and possibly the presence of a smart irrigation system.
  - The wine cellars and outbuildings are typical of a vineyard property and may have
    specialized climate control systems.
  area_devices:
    Dining Room:
    - light
    - smart_speaker
    Kitchen:
    - light
    - oven
    Courtyard:
    - pool_controller
    - outdoor_light
    Garden:
    - sprinkler
    Wine Cellar:
    - climate_control
  other_devices: []
device_entities:
- Dining Room:
  - name: light
    entities:
    - light.dining_room
  - name: smart_speaker
    entities:
    - media_player.dining_room
- Kitchen:
  - name: light
    entities:
    - light.kitchen
  - name: oven
    entities:
    - switch.kitchen_oven
- Courtyard:
  - name: pool_controller
    entities:
    - switch.pool_controller
  - name: outdoor_light
    entities:
    - light.courtyard_outdoor
- Garden:
  - name: sprinkler
    entities:
    - switch.garden_sprinkler
- Wine Cellar:
  - name: climate_control
    entities:
    - climate.wine_cellar
actions:
- Summarize the activities in the kitchen area
- Evaluate the current energy usage in the home
- Turn off all the lights in the bedroom area
- Notify the user if the front door has been left open for more than 10 minutes
- Check if the temperature in the living room is above the set threshold
- Determine if the garage door is closed at bedtime
- Notify the user if the humidity level in the bathroom is too high
- Evaluate the air quality in the home and suggest actions if it's poor
- Turn on the heating in the study room if the temperature drops below a certain level
- Notify the user if the security camera detects motion in the backyard
- Determine if the smart irrigation system needs to be activated based on weather
  forecast
- Check the status of the smart locks in the home
- Notify the user if the refrigerator door has been left open
- Summarize the electricity consumption pattern for the past week
- Evaluate if the home office area is experiencing poor air circulation
- Notify the user if the smoke alarm is triggered
- Analyze water usage data to identify potential leaks
- Determine if the garage lights have been left on during the day
- Notify the user if the pet feeder needs to be refilled
- Evaluate the overall security status of the home
