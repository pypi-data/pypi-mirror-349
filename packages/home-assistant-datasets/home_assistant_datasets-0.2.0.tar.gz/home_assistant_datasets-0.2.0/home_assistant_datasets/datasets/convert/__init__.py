"""Libraries for converting datasets between different formats."""

__all__ = [
    "intent",
]
