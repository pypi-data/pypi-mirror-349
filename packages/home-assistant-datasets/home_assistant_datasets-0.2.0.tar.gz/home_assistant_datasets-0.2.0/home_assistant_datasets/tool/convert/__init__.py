"""Leaderboard subcommand.

```
usage: home-assistant-datasets convert [-h] {intents} ...

positional arguments:
  {intents}    Sub Action

options:
  -h, --help  show this help message and exit
```
"""

from . import intents


SUBCMDS = {
    "intents": intents,
}
__all__ = list(SUBCMDS)
