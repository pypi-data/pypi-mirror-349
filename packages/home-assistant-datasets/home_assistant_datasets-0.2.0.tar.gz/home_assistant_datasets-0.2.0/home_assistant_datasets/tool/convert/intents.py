"""Convert intents tests into assist llm eval format for intents dataset.

```
usage: home-assistant-datasets convert intents [-h] --intents-repo-dir INTENTS_REPO_DIR
                                               --intents-dataset-dir INTENTS_DATASET_DIR

options:
  -h, --help            show this help message and exit
  --intents-repo-dir INTENTS_REPO_DIR
                        The path to the home-assistant/intents repo which is the source to read from.
  --intents-dataset-dir INTENTS_DATASET_DIR
                        The path to the home-assistant-datasets intents dataset directory to write to.
```
"""

import argparse
import logging
import pathlib
from tqdm import tqdm

from home_assistant_datasets.datasets.convert.intents import convert_intent_tests
from home_assistant_datasets.yaml_loaders import configure_encoders

__all__ = []

_LOGGER = logging.getLogger(__name__)

INTENTS_REPO_TEST_DIR = "tests"


def create_arguments(args: argparse.ArgumentParser) -> None:
    """Get parsed passed in arguments."""
    args.add_argument(
        "--intents-repo-dir",
        type=str,
        required=True,
        help="The path to the home-assistant/intents repo which is the source to read from.",
    )
    args.add_argument(
        "--intents-dataset-dir",
        type=str,
        required=True,
        help="The path to the home-assistant-datasets intents dataset directory to write to.",
    )


def run(args: argparse.Namespace) -> int:
    """Run the command line action."""

    intents_repo_dir = pathlib.Path(args.intents_repo_dir) / INTENTS_REPO_TEST_DIR
    if not intents_repo_dir.exists():
        raise ValueError(
            f"Path --intents-repo-dir did not point to repo, file {intents_repo_dir} does not exist"
        )
    intents_dataset_dir = pathlib.Path(args.intents_dataset_dir)

    _LOGGER.debug("Creating destination directory %s", str(intents_dataset_dir))
    intents_dataset_dir.mkdir(exist_ok=True)

    pbar = tqdm()
    configure_encoders()
    convert_intent_tests(intents_repo_dir, intents_dataset_dir, pbar)
    return 0
