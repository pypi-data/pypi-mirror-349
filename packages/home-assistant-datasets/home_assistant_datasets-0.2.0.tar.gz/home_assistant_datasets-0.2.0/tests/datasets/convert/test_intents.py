"""Test library for converting intents."""

from collections.abc import Generator
import datetime
import pathlib
import tempfile
from unittest.mock import patch

import pytest
from syrupy import SnapshotAssertion

from home_assistant_datasets.datasets.convert import intents

TESTDATA = pathlib.Path("tests/datasets/convert/testdata/intents")


@pytest.fixture(name="tmpdir")
def tmpdir_fixture() -> Generator[pathlib.Path]:
    """Fixture to produce a temporary directory."""
    with tempfile.TemporaryDirectory() as tmpdir:
        yield pathlib.Path(tmpdir)


@pytest.mark.parametrize(
    ("filename", "expected_domain", "expected_intent"),
    [
        ("cover_HassTurnOn.yaml", "cover", "HassTurnOn"),
        (
            "shopping_list_HassShoppingListAddItem.yaml",
            "shopping_list",
            "HassShoppingListAddItem",
        ),
    ],
)
def test_fixture_filename_parts(
    filename: pathlib.Path, expected_domain: str, expected_intent: str
) -> None:
    """Tests for fixture_filename_parts."""
    assert intents.fixture_filename_parts(pathlib.Path(filename)) == (
        expected_domain,
        expected_intent,
    )


def test_convert_intent_tests(
    tmpdir: pathlib.Path, snapshot: SnapshotAssertion
) -> None:
    """Test converting the intent tests to assist datasets."""
    FIXED_DATE = datetime.datetime(2025, 4, 20, 17, 20)

    with patch("home_assistant_datasets.datasets.convert.intents.now") as mock_now:
        mock_now.return_value = FIXED_DATE
        intents.convert_intent_tests(TESTDATA, tmpdir)
    found_files = [file for file in sorted(list(tmpdir.glob("**"))) if not file.is_dir()]
    assert [file.relative_to(tmpdir) for file in found_files] == snapshot

    for found_file in found_files:
        assert found_file.read_text() == snapshot
