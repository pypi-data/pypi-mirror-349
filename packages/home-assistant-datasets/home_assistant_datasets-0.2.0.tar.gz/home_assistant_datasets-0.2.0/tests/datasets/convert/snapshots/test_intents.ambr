# serializer version: 1
# name: test_convert_intent_tests
  list([
    PosixPath('dataset_card.yaml'),
    PosixPath('en/_fixtures.yaml'),
    PosixPath('en/cover_HassTurnOn.yaml'),
  ])
# ---
# name: test_convert_intent_tests.1
  '''
  ---
  description: '
  
    A dataset built form the Home Assistant intents repo, modeled after existing
  
    NLP test cases for the assistant pipeline.
  
  
    This dataset has been adapted for evaluating the Home Assistant LLM API ([blog post](https://developers.home-assistant.io/blog/2024/05/20/llm-api/))
  
    imported from the home assistant intents repo.
  
  
    This is mean to reuse the tests that already exist for the NLP, which turns out
  
    to expose some weaknesses or differences of interpretation of tasks. It also is
  
    a very large home which is challenging for smaller models given the ~100 or so
  
    devices. Lastly, there are some tests that have subtle mismatches that are reasonable
    (e.g.
  
    "minimum brightness" tests)
  
  
    This dataset uses the `assist` format. See the `assist` dataset card and README
    for
  
    additional details about the format and information about running the evaluation.
  
  
    Updated: 2025-04-20T17:20:00
  
    '
  name: intents
  urls:
  - https://github.com/allenporter/home-assistant-datasets/tree/main/datasets/intents
  - https://github.com/allenporter/home-assistant-datasets/tree/main/datasets/assist
  - https://github.com/home-assistant/intents
  
  '''
# ---
# name: test_convert_intent_tests.2
  '''
  ---
  language: en
  areas:
  - name: Guest Room
    id: guest_room_id
    floor: upstairs_id
  devices: []
  entities:
  - name: Shade Left
    id: cover.shade_left
    area: guest_room_id
    state: open
    attributes:
      device_class: shade
  
  '''
# ---
# name: test_convert_intent_tests.3
  '''
  ---
  category:
  - cover
  tests:
  - sentences:
    - open the garage door
    expect_response: Opened the garage
    expect_tool_call:
      tool_name: HassTurnOn
      tool_args:
        domain: cover
        device_class: garage
  
  '''
# ---
